core/src/main/java/org/apache/calcite/sql/type/SqlTypeAssignmentRules.java

```diff
@@ -207,262 +201,26 @@ private SqlTypeAssignmentRules(
-        coerceRules.copyValues(SqlTypeName.TIME)
-            .add(SqlTypeName.TIME)
-            .add(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.TIMESTAMP)
-            .add(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.CHAR)
-            .add(SqlTypeName.VARCHAR)
-            .addAll(SqlTypeName.BINARY_TYPES)
-            .build());
-
-    // TIME WITH LOCAL TIME ZONE is castable from...
-    coerceRules.add(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE,
-        coerceRules.copyValues(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.TIME)
-            .add(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.TIMESTAMP)
-            .add(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.CHAR)
-            .add(SqlTypeName.VARCHAR)
-            .addAll(SqlTypeName.BINARY_TYPES)
-            .build());
-
-    // TIMESTAMP is castable from...
-    coerceRules.add(SqlTypeName.TIMESTAMP,
-        coerceRules.copyValues(SqlTypeName.TIMESTAMP)
-            .add(SqlTypeName.TIMESTAMP)
-            .add(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.DATE)
-            .add(SqlTypeName.TIME)
-            .add(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.CHAR)
-            .add(SqlTypeName.VARCHAR)
-            .addAll(SqlTypeName.BINARY_TYPES)
-            .addAll(SqlTypeName.NUMERIC_TYPES)
-            .add(SqlTypeName.BOOLEAN)
-            .build());
-
-    // TIMESTAMP WITH LOCAL TIME ZONE is castable from...
-    coerceRules.add(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE,
-        coerceRules.copyValues(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.TIMESTAMP)
-            .add(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.DATE)
-            .add(SqlTypeName.TIME)
-            .add(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE)
-            .add(SqlTypeName.CHAR)
-            .add(SqlTypeName.VARCHAR)
-            .addAll(SqlTypeName.BINARY_TYPES)
-            .addAll(SqlTypeName.NUMERIC_TYPES)
-            .add(SqlTypeName.BOOLEAN)
-            .build());
-
     INSTANCE = new SqlTypeAssignmentRules(rules.map);
-    COERCE_INSTANCE = new SqlTypeAssignmentRules(coerceRules.map);
   }
 
   //~ Methods ----------------------------------------------------------------
 
-  /** Returns an instance that does not coerce. */
-  public static synchronized SqlTypeAssignmentRules instance() {
-    return instance(false);
-  }
-
   /** Returns an instance. */
-  public static synchronized SqlTypeAssignmentRules instance(boolean coerce) {
-    return coerce ? COERCE_INSTANCE : INSTANCE;
-  }
-
-  @Deprecated
-  public boolean canCastFrom(
-      SqlTypeName to,
-      SqlTypeName from,
-      boolean coerce) {
-    return instance(coerce).canCastFrom(to, from);
+  public static synchronized SqlTypeAssignmentRules instance() {
```

Comment:
why synchronized?

Reply:
The `synchronized ` keyword was always there, but i think we can remove it.